File: tablebenchmark/test/test_extract_segment.py
from pathlib import Path
import pandas as pd
import time

import sys
lib_path = Path("../src").resolve()
print(lib_path)
sys.path.append( lib_path.as_posix() )
import extract_segment

def show_t(t0, t1):
    print("  %.2f seconds" % (t1 - t0))

gtfs_dir = Path("../data/google_transit_manhattan").resolve()
print(gtfs_dir)

print("Open `shapes.txt`")
t0 = time.time()
frm_shape = pd.read_csv(gtfs_dir.joinpath("shapes.txt").as_posix(), dtype={'shape_id': str})
t1 = time.time()
show_t(t0, t1)

## get stop ids of shape
print("Open `stop_times.txt`")
t0 = time.time()
frm_stop_time = pd.read_csv(gtfs_dir.joinpath("stop_times.txt").as_posix(), dtype={'stop_id': str, 'shape_id': str})
t1 = time.time()
show_t(t0, t1)

print("Open `trips.txt`")
t0 = time.time()
frm_trip = pd.read_csv(gtfs_dir.joinpath("trips.txt").as_posix(), dtype={'trip_id': str, 'shape_id': str})
t1 = time.time()
show_t(t0, t1)

print("Open `stops.txt`")
t0 = time.time()
frm_stop = pd.read_csv(gtfs_dir.joinpath("stops.txt").as_posix(), dtype={'stop_id': str})
t1 = time.time()
show_t(t0, t1)

print("The running time of processing 500 rows")
t0 = time.time()
segements = {}
extract_segment.extract_segment(segements, frm_trip, frm_stop, frm_stop_time, frm_shape)
t1 = time.time()
show_t(t0, t1)

File: tablebenchmark/test/test_extract_segment.jl
using JuliaDB
using Test
include("../src/extract_segment.jl")

@testset "Julia Version RunningTime without precompile" begin
    gtfs_dir = "../data/google_transit_manhattan"

    @info "---------------Julia without precompile"
    @info "Check the time of loading CSV files"
    dtype=Dict("shape_id"=>String)
    @info "Open `shapes.txt`"
    @time frm_shape = loadtable(joinpath(gtfs_dir, "shapes.txt"), colparsers=dtype)
    @info "Open `stop_times.txt`"
    @time frm_stop_time = loadtable( joinpath(gtfs_dir, "stop_times.txt"), colparsers=Dict("stop_id"=>String, "shape_id"=>String) )
    @info "Open `trips.txt`"
    @time frm_trip = loadtable( joinpath(gtfs_dir, "trips.txt"), colparsers=Dict("trip_id"=>String, "shape_id"=>String) )
    @info "Open `stops.txt`"
    @time frm_stop = loadtable( joinpath(gtfs_dir, "stops.txt"), colparsers=Dict("stop_id"=>String) )

    segements = Dict{Tuple{String, String},Vector{Tuple{Float64, Float64}}}()
    @info "The running time of processing 500 rows:"
    @time extract_segment(segements, frm_trip, frm_stop, frm_stop_time, frm_shape)
end

@testset "Julia Version RunningTime (precompile)" begin
    gtfs_dir = "../data/google_transit_manhattan"

    @info "---------------Julia with precompile"
    @info "Check the time of loading CSV files"
    dtype=Dict("shape_id"=>String)
    @info "Open `shapes.txt`"
    loadtable(joinpath(gtfs_dir, "shapes.txt"), colparsers=dtype)
    @time frm_shape = loadtable(joinpath(gtfs_dir, "shapes.txt"), colparsers=dtype)
    @info "Open `stop_times.txt`"
    loadtable( joinpath(gtfs_dir, "stop_times.txt"), colparsers=Dict("stop_id"=>String, "shape_id"=>String) )
    @time frm_stop_time = loadtable( joinpath(gtfs_dir, "stop_times.txt"), colparsers=Dict("stop_id"=>String, "shape_id"=>String) )
    @info "Open `trips.txt`"
    loadtable( joinpath(gtfs_dir, "trips.txt"), colparsers=Dict("trip_id"=>String, "shape_id"=>String) )
    @time frm_trip = loadtable( joinpath(gtfs_dir, "trips.txt"), colparsers=Dict("trip_id"=>String, "shape_id"=>String) )
    @info "Open `stops.txt`"
    loadtable( joinpath(gtfs_dir, "stops.txt"), colparsers=Dict("stop_id"=>String) )
    @time frm_stop = loadtable( joinpath(gtfs_dir, "stops.txt"), colparsers=Dict("stop_id"=>String) )

    segements = Dict{Tuple{String, String},Vector{Tuple{Float64, Float64}}}()

    @info "The running time of processing 500 rows:"
    extract_segment(segements, frm_trip, frm_stop, frm_stop_time, frm_shape, testing=true)
    @time extract_segment(segements, frm_trip, frm_stop, frm_stop_time, frm_shape)
end
